platforms/posix/src/px4_daemon/server.cpp

```diff
@@ -135,14 +135,11 @@ Server::_server_main()
 		return;
 	}
 
-	// The list of file descriptors to watch.
-	std::vector<pollfd> poll_fds;
-
 	// Watch the listening socket for incoming connections.
-	poll_fds.push_back(pollfd {_fd, POLLIN, 0});
+	pollfd poll_fd {_fd, POLLIN, 0};
 
 	while (true) {
-		int n_ready = poll(poll_fds.data(), poll_fds.size(), -1);
+		int n_ready = poll(&poll_fd, 1, -1);
```

Comment:
(By the way, polling for a single file descriptor doesn't make much sense. If you really want wait for only this single fd, just call `accept()` on it directly, which will do the blocking/waiting.)